In [53]:
import pandas as pd
import numpy as np
import plotly.express as px

In [54]:
heart = pd.read_csv('Data/hd.csv')
heart

age  sex   cp  trestbps   chol  fbs  restecg  \
0                               52  1.0  0.0     125.0  212.0  0.0      1.0   
1                               53  1.0  0.0     140.0  203.0  1.0      0.0   
2                               70  1.0  0.0     145.0  174.0  0.0      1.0   
3                               61  1.0  0.0     148.0  203.0  0.0      1.0   
4                               62  0.0  0.0     138.0  294.0  1.0      1.0   
...                            ...  ...  ...       ...    ...  ...      ...   
1021                            60  1.0  0.0     125.0  258.0  0.0      0.0   
1022                            47  1.0  0.0     110.0  275.0  0.0      0.0   
1023                            50  0.0  0.0     110.0  254.0  0.0      0.0   
1024                            54  1.0  0.0     120.0  188.0  0.0      1.0   
1025  <pre></pre><strong></strong>  NaN  NaN       NaN    NaN  NaN      NaN   

      thalach  exang  oldpeak  slope   ca  thal  target  
0       168.0    0.0      1.0    2.0  2.0   3.0     0.0  
1       155.0    1.0      3.1    0.0  0.0   3.0     0.0  
2       125.0    1.0      2.6    0.0  0.0   3.0     0.0  
3       161.0    0.0      0.0    2.0  1.0   3.0     0.0  
4       106.0    0.0      1.9    1.0  3.0   2.0     0.0  
...       ...    ...      ...    ...  ...   ...     ...  
1021    141.0    1.0      2.8    1.0  1.0   3.0     0.0  
1022    118.0    1.0      1.0    1.0  1.0   2.0     0.0  
1023    159.0    0.0      0.0    2.0  0.0   2.0     1.0  
1024    113.0    0.0      1.4    1.0  1.0   3.0     0.0  
1025      NaN    NaN      NaN    NaN  NaN   NaN     NaN  

[1026 rows x 14 columns]

In [55]:
heart.columns

Index(['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach',
       'exang', 'oldpeak', 'slope', 'ca', 'thal', 'target'],
      dtype='object')

In [56]:
heart.shape

(1026, 14)

column info:

age

sex

cp\- chest pain type \(4 values\)

trestbps\- resting blood pressure

chol\- serum cholestoral in mg/dl

fbs\- fasting blood sugar &gt; 120 mg/dl

thalach\- resting electrocardiographic results \(values 0,1,2\)

exang\- maximum heart rate achieved

oldexercise induced angina

oldpeak = ST depression induced by exercise relative to rest

slope\-the slope of the peak exercise ST segment

ca\- number of major vessels \(0\-3\) colored by flourosopy

thal: 0 = normal; 1 = fixed defect; 2 = reversable defect



In [57]:
heart.dropna(inplace=True)
heart.reset_index(drop=True, inplace=True)

In [58]:
len(heart)

1025

In [59]:
heart.age.head()

0    52
1    53
2    70
3    61
4    62
Name: age, dtype: object

In [60]:
fig = px.histogram(heart, x="age")
fig.show()

In [61]:
fig = px.scatter(heart, x="trestbps")
fig.show()

In [62]:
heart.corr()

sex        cp  trestbps      chol       fbs   restecg  \
sex       1.000000 -0.041119 -0.078974 -0.198258  0.027200 -0.055117   
cp       -0.041119  1.000000  0.038177 -0.081641  0.079294  0.043581   
trestbps -0.078974  0.038177  1.000000  0.127977  0.181767 -0.123794   
chol     -0.198258 -0.081641  0.127977  1.000000  0.026917 -0.147410   
fbs       0.027200  0.079294  0.181767  0.026917  1.000000 -0.104051   
restecg  -0.055117  0.043581 -0.123794 -0.147410 -0.104051  1.000000   
thalach  -0.049365  0.306839 -0.039264 -0.021772 -0.008866  0.048411   
exang     0.139157 -0.401513  0.061197  0.067382  0.049261 -0.065606   
oldpeak   0.084687 -0.174733  0.187434  0.064880  0.010859 -0.050114   
slope    -0.026666  0.131633 -0.120445 -0.014248 -0.061902  0.086086   
ca        0.111729 -0.176206  0.104554  0.074259  0.137156 -0.078072   
thal      0.198424 -0.163341  0.059276  0.100244 -0.042177 -0.020504   
target   -0.279501  0.434854 -0.138772 -0.099966 -0.041164  0.134468   

           thalach     exang   oldpeak     slope        ca      thal    target  
sex      -0.049365  0.139157  0.084687 -0.026666  0.111729  0.198424 -0.279501  
cp        0.306839 -0.401513 -0.174733  0.131633 -0.176206 -0.163341  0.434854  
trestbps -0.039264  0.061197  0.187434 -0.120445  0.104554  0.059276 -0.138772  
chol     -0.021772  0.067382  0.064880 -0.014248  0.074259  0.100244 -0.099966  
fbs      -0.008866  0.049261  0.010859 -0.061902  0.137156 -0.042177 -0.041164  
restecg   0.048411 -0.065606 -0.050114  0.086086 -0.078072 -0.020504  0.134468  
thalach   1.000000 -0.380281 -0.349796  0.395308 -0.207888 -0.098068  0.422895  
exang    -0.380281  1.000000  0.310844 -0.267335  0.107849  0.197201 -0.438029  
oldpeak  -0.349796  0.310844  1.000000 -0.575189  0.221816  0.202672 -0.438441  
slope     0.395308 -0.267335 -0.575189  1.000000 -0.073440 -0.094090  0.345512  
ca       -0.207888  0.107849  0.221816 -0.073440  1.000000  0.149014 -0.382085  
thal     -0.098068  0.197201  0.202672 -0.094090  0.149014  1.000000 -0.337838  
target    0.422895 -0.438029 -0.438441  0.345512 -0.382085 -0.337838  1.000000

In [63]:
correlation= heart.corr()

In [64]:
fig = px.imshow(correlation, text_auto = True)
fig.update_layout(
    width=900,
    height=900,
    title_text = 'Correlation Heatmap of Data'
)
fig.show()

This scatter plot shows the resting blood pressure of a person according to their age. The color of the dot represents their sex, and its size represents their serum cholesterol.



In [65]:
fig = px.scatter(heart, x="trestbps", y="age", color="sex",
                 size='chol')
fig.show()

In [72]:

fig = px.scatter(heart, x='age', y='exang', opacity=0.65, trendline='ols', trendline_color_override='darkblue')

fig.show()

ModuleNotFoundError: No module named 'statsmodels'